src/Core/AbstractTest.php

```diff
@@ -182,6 +182,30 @@ public function setPending($state)
     }
 
     /**
+     * Set the focused status of the test and its children according to the
+     * supplied focus pattern and/or skip pattern
+     *
+     * @param string|null $focusPattern
+     * @param string|null $skipPattern
+     */
+    public function applyFocusPatterns($focusPattern, $skipPattern = null)
+    {
```

Comment:
Just to double check, we are saying `$skipPattern` and `$focusPattern` are not mutually exclusive right?

You could run a test that has a combination of a focus pattern and a skip pattern? Like "run integration tests, but skip the selenium ones?"